File: tests/PHP/CaesarTest.php
<?php
include '../../lib/PHP/Caesar.php';

class CaesarTest extends PHPUnit_Framework_TestCase
{
    // ...

    public function testSanityCheck()
    {
        $c = new Caesar(3);
        $this->assertEquals("khoor", $c->encrypt("hello"));
        $this->assertEquals("hello", $c->decrypt("khoor"));
        $e = new Exploits();
         //$e->ciphertext_only("khoor");
        $this->assertEquals(3, $e->known_plaintext("hello", "khoor"));
        $this->assertEquals(3, $e->chosen_plaintext($c));
        $this->assertEquals(3, $e->chosen_ciphertext($c));

    }
}
?>

File: tests/Ruby/CaesarTest.rb
require '../../lib/Ruby/Caesar.rb'
require 'test/unit'

class CaesarTest < Test::Unit::TestCase

    def test_sanity
        c = Caesar.new(3)
        assert_equal("khoor", c.encrypt("hello"))
        assert_equal("hello", c.decrypt("khoor"))
        e = Exploits.new()
        #e.ciphertext_only("khoor")
        assert_equal(3, e.known_plaintext("hello","khoor"))
        assert_equal(3, e.chosen_plaintext(c))
        assert_equal(3, e.chosen_ciphertext(c))
    end
end
